packages/botonic-plugin-flow-builder/src/user-input/smart-intent.ts

```diff
@@ -4,66 +4,61 @@ import axios from 'axios'
 import { FlowBuilderApi } from '../api'
 import { HtSmartIntentNode } from '../content-fields/hubtype-fields/smart-intent'
 
-interface InferenceParam {
-  name: string
-  definition: string
+export interface SmartIntentsInferenceParams {
+  bot_id: string
+  text: string
+  use_latest: boolean
+  num_smart_intents_to_use?: number
 }
 export class SmartIntentsApi {
   public cmsApi: FlowBuilderApi
   public currentRequest: ActionRequest
+  public smartIntentsConfig: Partial<SmartIntentsInferenceParams>
 
-  constructor(cmsApi: FlowBuilderApi, request: ActionRequest) {
+  constructor(
+    cmsApi: FlowBuilderApi,
+    request: ActionRequest,
+    smartIntentsConfig: Partial<SmartIntentsInferenceParams>
+  ) {
     this.currentRequest = request
     this.cmsApi = cmsApi
+    this.smartIntentsConfig = smartIntentsConfig
   }
 
   async getNodeByInput(): Promise<HtSmartIntentNode | undefined> {
+    if (!this.currentRequest.input.data) return undefined
```

Comment:
When you get into this action you will always have input.data, do you need to do this so typescript doesn't say it can be undefined on line 35?